### DISCUSSION TITLE
Kenapa output yang di hasilkan strtr(To::JSON([...])) null, saat content mengandung element url/ link ?

### DISCUSSION THREAD

**Discussion by lokersumatera** [mecha-cms/mecha#199] [Q&A]
Kang, pada panel input text yang mengandung elemen `<a href="alamaturl.com/link">URL</a>`, saat ditampilkan sebagai struktur data JSON outputnya null, saat saya hapus elemen yang mengandung url/ link outputnya normal.

Berikut kode yang saya tulis:

``` php
<script type="application/ld+json">
<?=strtr(To::JSON([
...
'description'=>$page->job_description
...
]), ["\\/" => '/']);?>
</script>
```

**Comment by taufik-nurrohman** [comment]
Yang `null` keseluruhan halaman atau cuma di dalam tag `<script type="application/ld+json">` saja?

**Reply by lokersumatera** [reply]
``` html
<script type="application/ld+json">
null
</script>
```
yang `null` hanya di dalam tag aja kang

**Reply by taufik-nurrohman** [reply]
Coba mungkin sintaks _array_ ada yang tidak valid. Kalau diberi input yang lebih sederhana hasilnya bagaimana? Misal begini:

~~~ php
<?= To::JSON([1, 2, 'a']); ?>
~~~

**Reply by lokersumatera** [reply]
```
<!-- Contoh Sederhana -->
<?= To::JSON([1, 2, 'a']); ?>
<!-- Contoh dikasih url -->
<?= To::JSON([1, 2, '<a href="https://lokersumatera.my.id/lowongan-kerja/jendela-coffee" rel="nofollow">url</a>']); ?>
<!--Contoh dikasih url eksternal -->
<?= To::JSON([1, 2, '<a href="https://www.instagram.com/jendelacoffee.pekanbaru/">url</a>']); ?>
```

Output:
```
<!-- Contoh Sederhana -->
[1,2,"a"]

<!-- Contoh dikasih url -->
[1,2,"<a href="https://lokersumatera.my.id/lowongan-kerja/1" rel>url<\/a>"]

<!--Contoh dikasih url eksternal -->
[1,2,"<a href="https://lokersumatera.my.id/lowongan-kerja/1">url<\/a>"]
```

**Reply by taufik-nurrohman** [reply]
Lha itu jadi semua. Berarti masalahnya ada di input JSON yang tidak valid.

**Reply by lokersumatera** [reply]
wah, gimana tuh kang?. urlnya pun bisa berubah gitu juga

**Reply by taufik-nurrohman** [reply]
Coba ekstensi `link` dicopot dulu.

**Reply by lokersumatera** [reply]
udah normal kang, setelah ekstensi link dicopot

**Reply by taufik-nurrohman** [reply]
Oke ini _bug_ berarti.

**Reply by taufik-nurrohman** [reply]
Coba edit bagian ini:

https://github.com/mecha-cms/mecha/commit/3e251e16c0dd0841cc15413ebdfcd5e1a8cac849

**Reply by lokersumatera** [reply]
done kang

**Reply by taufik-nurrohman** [reply]
Tapi masih belum jadi kayaknya ini.
